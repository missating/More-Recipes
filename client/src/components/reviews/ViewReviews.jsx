import React from 'react';
import PropTypes from 'prop-types';
import Gravatar from 'react-gravatar';

const ViewReviews = props => (
  <div className="container">
    <div className="image">
      <Gravatar
        className="img rounded"
        email={props.email}
      />
    </div>
    <p> {props.content} </p>
    <h5> <em> {props.user}</em> </h5>
    <hr />
  </div>
);

ViewReviews.propTypes = {
  content: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired
};

export default ViewReviews;
