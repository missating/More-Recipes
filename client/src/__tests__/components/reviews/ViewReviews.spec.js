import React from 'react';
import toJSON from 'enzyme-to-json';
import ViewReviews from '../../../components/reviews/ViewReviews';

describe('view Reviews component', () => {
  const props = {
    content: 'good stuff',
    user: 'test',
    email: 'test@test.com'
  };

  it('Should render', () => {
    const wrapper = shallow(<ViewReviews
      {...props}
    />);
    expect(wrapper.find('div').length).toEqual(2);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

