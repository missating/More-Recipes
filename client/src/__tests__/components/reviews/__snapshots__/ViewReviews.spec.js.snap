// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view Reviews component Should render 1`] = `
<div
  className="container"
>
  <div
    className="image"
  >
    <Gravatar
      className="img rounded"
      default="retro"
      email="test@test.com"
      protocol="//"
      rating="g"
      size={50}
    />
  </div>
  <p>
     
    good stuff
     
  </p>
  <h5>
     
    <em>
       
      test
    </em>
     
  </h5>
  <hr />
</div>
`;
